pkg/pillar/cmd/zedrouter/acl.go

```diff
@@ -1580,13 +1581,25 @@ func createMarkAndAcceptChain(aclArgs types.AppNetworkACLArgs,
 		return errors.New("Invalid chain creation")
 	}
 
+	chainFlush := []string{"-t", "mangle", "--flush", name}
+
 	newChain := []string{"-t", "mangle", "-N", name}
 	log.Functionf("createMarkAndAcceptChain: Creating new chain (%s)", name)
 	err := iptables.IptableCmd(log, newChain...)
 	if err != nil {
-		log.Errorf("createMarkAndAcceptChain: New chain (%s) creation failed: %s",
+		// if chain already exists, we can skip this error
+		if !strings.Contains(err.Error(), "Chain already exists") {
+			log.Errorf("createMarkAndAcceptChain: New chain (%s) creation failed: %s",
+				name, err)
+			return err
+		}
+		log.Warningf("createMarkAndAcceptChain: New chain (%s) creation failed skipped: %s",
 			name, err)
-		return err
+		if err := iptables.IptableCmd(log, chainFlush...); err != nil {
```

Comment:
With old code, this function returned with error when the chain already exists. And that worked because the marking value (which is nothing but the acl rule number) is embedded into the chain name. So, if the chain name does not change, marking value would not change. We may not require flushing and adding rules again.

Reply:
Yes, you are right, but in the attached iptables rules you can see empty chain `bn1-nbu1x2-1` for example.

Reply:
Empty rule chain sounds like a new problem. There shouldn't be a chain with no rules. Flushing and adding rules gain may solve that problem, but we need to understand the reason an empty chain exists. What case do we flush the chain, but not delete the chain itself? Or does it happen when app instance is deleted (we flush chain and hit error when chain delete is executed)?

Reply:
Yes, it is after delete-create cycle for apps (but I send here rules after trying to create app again). And I see `Too many links` message, so we have -j to it and cannot delete chain.

Reply:
What should I do to reproduce the problem where there are empty rule chains?

Reply:
It is quite difficult to reproduce my tests, please try this variant:
```
git clone https://github.com/lf-edge/eden.git
cd eden
make clean
make run
./eden eve onboard
```

then repeat few times two of our tests:
```
./eden test tests/workflow -p eden.escript.test -r TestEdenScripts/2dockers_test -a '-testdata ../app/testdata/'
./eden test tests/workflow -p eden.escript.test -r TestEdenScripts/test_networking -a '-testdata ../network/testdata/'
```

You can see rules with going inside EVE by `./eden eve ssh` or `./eden eve console`. You can see applications with `./eden pod ps`.

Reply:
I was hoping for some simpler manual steps to reproduce the empty chain list problem. It might take some time for me to be able to try the steps above. The current fix may cover up the empty list issue, but we should address the empty list issue separately since empty lists are not expected.
May be you can go ahead with merging the current fix and look into the empty list problem separately.